lib/dialects/abstract/connection-manager.js

```diff
@@ -85,20 +90,7 @@ class ConnectionManager {
 
     if (!config.replication) {
       this.pool = Pooling.createPool({
-        create: () => new Promise(resolve => {
-          this
-            ._connect(config)
-            .tap(() => {
-              this.poolError = null;
-            })
-            .then(resolve)
-            .catch(e => {
-              // dont throw otherwise pool will release _dispense call
-              // which will call _connect even if error is fatal
-              // https://github.com/coopernurse/node-pool/issues/161
-              this.poolError = e;
-            });
-        }),
+        create: () => this._connect(config),
```

Comment:
I think converting <code>reject(err)</code> to <code>resolve(err)</code> should be **here**. This hack is only for pool. <code>_connect</code> shouldn't be hacky.

```javascript
this._connect(config).catch(err => err);
```